test/integration/smoke/test_routers.py

```diff
@@ -796,3 +830,122 @@ def test_09_reboot_router(self):
             "Router response after reboot is either is invalid\
                     or in stopped state")
         return
+
+    def check_virtual_machine_accessibility(self, virtual_machine):
+        response = virtual_machine.getState(
+            self.userapiclient,
+            VirtualMachine.RUNNING
+        )
+        self.assertEqual(response[0], PASS, "Virtual machine is not running")
+
+        # Check VM Accessibility
+        try:
+            SshClient(host=virtual_machine.ssh_ip,
+                      port=self.services['natrule']['publicport'],
+                      user=virtual_machine.username,
+                      passwd=virtual_machine.password)
+        except Exception as e:
+            raise Exception("Error while trying SSH to VM: %s" % e)
+
+    def try_ssh_to_host(self):
+        list_virtual_machine_response = list_virtual_machines(
+            self.apiclient,
+            listall=True)
+        status = validateList(list_virtual_machine_response)
+        self.assertEqual(
+            status[0],
+            PASS,
+            "VM List Validation Failed due to %s" %
+            status[2])
+
+        host_id = list_virtual_machine_response[0].hostid
+        list_host_response = list_hosts(self.apiclient, id=host_id)
+        status = validateList(list_host_response)
+        self.assertEqual(
+            status[0],
+            PASS,
+            "Host List Validation Failed due to %s" %
+            status[2])
+
+        self.services['hosts'][self.hypervisor.lower()]['password'] = "xenroot"
+        ssh_object = SshClient(
+            host=list_host_response[0].ipaddress,
+            port=self.services['natrule']['publicport'],
+            user=self.services['hosts'][self.hypervisor.lower()]['username'],
+            passwd=self.services['hosts'][self.hypervisor.lower()]['password']
+        )
+        hostname_from_host = ssh_object.execute("uname -n")
+        ssh_object.execute("ssh-keygen -R %s" % hostname_from_host[0])
+        ssh_object.execute("exit")
+        return ssh_object
```

Comment:
What is the guarantee that  RouterVM and Guest vm will be on the same host? We have to the host id of the VR and  get the ssh object for that host.